### PR TITLE
Variants GetVariantsMatch Service

### PR DESCRIPTION
**Pull request by bricht**
-Added GetVariantsMatch functionality
-Added api test

### Fixed Issues (if relevant)

1. Fixes magento/catalog-storefront#359

### Code Review Checklist (*)

See detailed [checklist](https://github.com/magento/catalog-storefront/blob/develop/dev/docs/projectAgreements/Code-Review-checklist.md)

- [ ] Story AC is completed
- [ ] proposed changes correspond to [Magento Technical Vision](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html)
- [ ] new or changed code is covered with web-api/integration tests (if applicable)
  - expected results in test verified with data from fixture
- [ ] no backward incompatible changes
- [ ] Export API (et_schema.xml) and SF API schemas (proto schema) are reflected in the codebase
  - prerequisite: story branch created with all needed generated classes according to proposes schema-changes
  - DTO classes do not contain any manual changes (Magento\CatalogExportApi\*, Magento\CatalogStorefrontApi\*)
- [ ] Class usage: magento/catalog-storefront repo don't use directly classes from magento/saas-export repo and vise-verse
  - Check composer.json dependencies
- [ ] Legacy code is deleted
  - Any Data Providers present in Connector part  (Magento\CatalogStorefrontConnector, Magento\*Extractor modules)
  - And Data Providers from Export API (magento/saas-export repo) that is not relevant anymore
  - Any DTO for Export API/SF API which does not reflect current schema: et_schema, proto schema
  - Any “mapper” on Message Broker (between Export API and SF API)
    - if mapper still needed, verify fields used in mapping, remove not relevant fields

### CODE REVIEW

**Comment by bricht** [comment]
@magento run all tests

**Comment by bricht** [comment]
@magento run all tests

File: app/code/Magento/CatalogStorefront/Test/Api/ProductVariants/ConfigurableVariantsTest.php [hunk 1]
```diff
@@ -109,19 +117,75 @@ public function testConfigurableProductVariants(): void
...
+            'simple_8'
+        ];
+        /** @var $configurable Product */
+        $configurable = $this->productRepository->get('configurable');
+        $simples = [];
+        foreach ($simpleSkus as $sku) {
+            $simples[] = $this->productRepository->get($sku);
+        }
+        $availableVariants = $this->getExpectedProductVariants($configurable, $simples);
+        self::assertCount(9, $availableVariants, 'Wrong number of expected variants');
```
Comment (RuslanKostiv1) [inline app/code/Magento/CatalogStorefront/Test/Api/ProductVariants/ConfigurableVariantsTest.php:165]: We don't test here our API, this part already been tested by Magento. If this error occurs "Wrong number of expected variants" we would guess we have an issue with API, but the problem will be with Magento repositories.

**Comment by bricht** [comment]
@magento run all tests

File: app/code/Magento/CatalogStorefront/Test/Api/ProductVariants/ConfigurableVariantsTest.php (hunk 1 above)
Comment (bricht) [reply app/code/Magento/CatalogStorefront/Test/Api/ProductVariants/ConfigurableVariantsTest.php]: Removed this part. We will get null pointer anyway if its empty, but I guess that's an error and not test failure.

**Comment by bricht** [comment]
@magento run Static Tests

**Comment by bricht** [comment]
@magento run Static Tests

**Comment by bricht** [comment]
@magento run all tests

**Comment by bricht** [comment]
@magento run all tests

**Comment by bricht** [comment]
@magento run all tests

**Comment by RuslanKostiv1** [comment]
@magento import pr to magento-commerce/catalog-storefront-ce

**Comment by magento-engcom-team** [comment]
@RuslanKostiv1 the pull request successfully imported.

**Pull request merged**